src/systems/systemDialog.component.test.tsx

```diff
@@ -90,6 +92,24 @@ describe('Systems Dialog', () => {
       expect(screen.getByText('Add Subsystem')).toBeInTheDocument();
     });
 
+    it('disabled button and shows circular progress indicator when request is pending', async () => {
```

Comment:
```suggestion
    it('disables save button and shows circular progress indicator when request is pending', async () => {
```

I think the edit tests should have this as well since this test would still pass if isEditPending was removed by accident, same for the others.